src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxIPGlobalProperties.cs

```diff
@@ -0,0 +1,362 @@
+            string[] v6connections = tcp6FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            TcpConnectionInformation[] connections = new TcpConnectionInformation[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // UDP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            // UDP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                connections[index++] = ParseTcpConnectionInformationFromLine(line);
+            }
+
+            return connections;
+        }
+
+        public override IPEndPoint[] GetActiveTcpListeners()
+        {
+            string tcp4FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp4ConnectionsFile);
+            string[] v4connections = tcp4FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            string tcp6FileContents = File.ReadAllText(LinuxNetworkFiles.Tcp6ConnectionsFile);
+            string[] v6connections = tcp6FileContents.Split(newLineSeparator, StringSplitOptions.RemoveEmptyEntries);
+
+            IPEndPoint[] endPoints = new IPEndPoint[v4connections.Length + v6connections.Length - 2]; // First line is header in each file
+            int index = 0;
+
+            // TCP Connections
+            for (int i = 1; i < v4connections.Length; i++) // Skip first line header
+            {
+                string line = v4connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            // TCP6 Connections
+            for (int i = 1; i < v6connections.Length; i++) // Skip first line header
+            {
+                string line = v6connections[i];
+                IPAddress remoteIPAddress;
+                int remotePort;
+                ParseRemoteConnectionInformation(line, out remoteIPAddress, out remotePort);
+
+                endPoints[index++] = new IPEndPoint(remoteIPAddress, remotePort);
+            }
+
+            return endPoints;
+        }
+
+        public override IPEndPoint[] GetActiveUdpListeners()
+        {
+            string udp4FileContents = File.ReadAllText(LinuxNetworkFiles.Udp4ConnectionsFile);
```

Comment:
1. I'm not sure about this. @stephentoub do you know if there's anything in particular to watch out for here, regarding Encodings and procfs files?

2. ReadAllText should just do a read-only open on the file.

3. I'm not sure about this either. I think that all of the files I'm using are generally supposed to be readable by anyone (the stuff under /proc/net/*), but I think there are other areas that you probably need special permissions to access, or at least write to, like the configuration files. I don't write to anything, so we should be okay here. @stephentoub, @pgavlin does the above sound accurate?